api/app/controllers/spree/api/address_books_controller.rb

```diff
@@ -2,32 +2,41 @@ module Spree
   module Api
     class AddressBooksController < Spree::Api::BaseController
       # Note: the AddressBook is the resource to think about here, not individual addresses
+      before_filter :authorize_user_address
+      before_filter :load_user_addresses
 
       def show
-        render_address_book
+        render :show, status: :ok
       end
 
       def update
         address_params = address_book_params
         default_flag = address_params.delete(:default)
-        address = current_api_user.save_in_address_book(address_params, default_flag)
+        address = address_book_user.save_in_address_book(address_params, default_flag)
         if address.valid?
-          render_address_book
+          render :show, status: :ok
         else
           invalid_resource!(address)
         end
       end
 
       def destroy
-        current_api_user.remove_from_address_book(params[:address_id])
-        render_address_book
+        address_book_user.remove_from_address_book(params[:address_id])
+        render :show, status: :ok
       end
 
       private
 
-      def render_address_book
-        @user_addresses = current_api_user.user_addresses
-        render :show, status: :ok
+      def authorize_user_address
+        authorize! :manage, UserAddress.new(user_id: address_book_user.id)
```

Comment:
A couple ideas for trying to avoid rewriting address book stuff significantly (just brainstorming here)
```ruby
# user_address_book.rb
module UserAddressBook
  class Permission
    attr_reader :user
    def initialize(user); @user = user; end
  end
end
# default_customer.rb
can :manage, UserAddressBook::Permission, user: user
# address_books_controller.rb
authorize! :manage, UserAddressBook::Permission.new(address_book_user)
```
or another idea:
```
# default_customer.rb
can manage_address_book, User, id: user.id
# address_books_controller.rb
authorize! :manage_address_book, address_book_user
```